MyLibrary/Tests/MyLibraryTests/MyLibraryTests.swift

```diff
@@ -68,5 +69,38 @@ final class MyLibraryTests: XCTestCase {
         // Then
         XCTAssertNil(isLuckyNumber)
     }
+    
+    func testWeatherModelForValidTemperature() {
+        let decoder = JSONDecoder()
+        let data = Data("{\"main\": {\"temp\":58}}".utf8)
+        do {
+            let container = try decoder.decode(Weather.self, from: data)
+            XCTAssertNotNil(container)
+            XCTAssert(container.main.temp == 58.0)
+        } catch {
+            XCTFail("Failed to decode temp from json")
+        }
+    }
```

Comment:
You don't need a try-catch block.  You could instead just have the test function throw:
```
func testWeatherModelForValidTemperature() throws {`
```
and the test will automatically fail if it throws.

Also, although not everyone is a fan, I would encourage you to stick with "Given / When / Then" (see the test above for an example).